dist/common/scripts/scylla-housekeeping

```diff
@@ -80,8 +80,16 @@ def get_api(path):
     return get_json_from_url("http://" + api_address + path)
 
 
+def parse_scylla_version(version):
+    # Newer setuptools does not accept ~dev in version strings, need to
+    # replace it to .dev0
+    if version and version.endswith('~dev'):
```

Comment:
you still didn't answered me, if this won't be failing for `6.1.0~rc1` for example ?

Reply:
looks at
https://github.com/scylladb/scylla-ccm/blob/next/ccmlib/utils/version.py

and it's testing:
https://github.com/scylladb/scylla-ccm/blob/next/tests/test_utils_version.py

for possible other variant you'll need to consider

Reply:
I think you only need `s/~/-/`, see:
```python
Python 3.12.1 (main, Jan  8 2024, 23:17:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from packaging.version import parse, Version
>>> parse('6.1.0~rc1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fruch/.pyenv/versions/cassandra-3.12.1/lib/python3.12/site-packages/packaging/version.py", line 54, in parse
    return Version(version)
           ^^^^^^^^^^^^^^^^
  File "/home/fruch/.pyenv/versions/cassandra-3.12.1/lib/python3.12/site-packages/packaging/version.py", line 200, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '6.1.0~rc1'
>>> parse('6.1.0-rc1')
<Version('6.1.0rc1')>
```

Reply:
s/~/-/ doesn't give the same ordering. ~ means before, 1~a is before 1, while 1-a is after 1.

Reply:
Claude suggested this:

```python
from debian.debian_support import version_compare as cmp_ver
result = cmp_ver('1.2.3-1', '1.2.3-2')
```

Reply:
I convinced Claude to write this  (untested):

```python
from functools import cmp_to_key

def compare_debian_versions(a, b):
    def parse_version(v):
        epoch = v.split(':')[0] if ':' in v else '0'
        version = v.split(':')[-1].split('-')[0] if '-' in v else v.split(':')[-1]
        parts = [int(x) if x.isdigit() else x for x in version.split('.')]
        return (int(epoch), parts)

    return (parse_version(a) > parse_version(b)) - (parse_version(a) < parse_version(b))

```

Reply:
It's for a warning for the user when installing Scylla 

Not sure all it needs to be that accurate in that sense.

Also I've referenced the implementation we are using and testing in CCM and in SCT. (Clude wasn't available at the time)

Reply:
Sent fix for ~rcN versions. As @fruch pointed out, it seems like parse_version() automatically convert to Python versioning scheme when we replace "~" to "-":
6.0.0-rc3 will automatically replace to 6.0.0rc3
6.1.0-dev will automatically replace to 6.1.0.dev0

But I'm not sure this conversion will keep in future versions, I think it's safer to follow Python versioning scheme, which is X.Y.Z.devN for dev versions and X.Y.ZrcN for rc versions.